buildpacks/go/faas/main.go

```diff
@@ -47,58 +50,69 @@ func run() error {
 		cancel()
 	}()
 
-	var handler interface{} = function.Handle
+	// Use a gorilla mux for handling all HTTP requests
+	router := mux.NewRouter()
 
-	httpHandler := toHttpHandler(handler, ctx)
+	// Add handlers for readiness and liveness endpoints
+	router.HandleFunc("/health/{endpoint:readiness|liveness}", func(w http.ResponseWriter, r *http.Request) {
+		json.NewEncoder(w).Encode(map[string]bool{"ok": true})
+	})
 
-	if httpHandler != nil {
-		httpServer := &http.Server{
-			Addr:           fmt.Sprintf(":%d", *port),
-			Handler:        httpHandler,
-			ReadTimeout:    1 * time.Minute,
-			WriteTimeout:   1 * time.Minute,
-			MaxHeaderBytes: 1 << 20,
-		}
+	var handler interface{} = function.Handle
 
-		listenAndServeErr := make(chan error, 1)
-		go func() {
-			if *verbose {
-				fmt.Printf("listening on http port %v for HTTP requests\n", *port)
-			}
-			err := httpServer.ListenAndServe()
-			cancel()
-			listenAndServeErr <- err
-		}()
-
-		<- ctx.Done()
-		shutdownCtx, shutdownCancelFn := context.WithTimeout(context.Background(), time.Second * 5)
-		defer shutdownCancelFn()
-		err := httpServer.Shutdown(shutdownCtx)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "error on server shutdown: %v\n", err)
-		}
+	httpHandler := toHttpHandler(handler, ctx)
 
-		err = <-listenAndServeErr
-		if http.ErrServerClosed == err {
-			return nil
+	if httpHandler == nil {
+		if *verbose {
+			fmt.Printf("Initializing CloudEvent function\n")
 		}
-		return err
-	} else {
-		transport, err := cloudevents.NewHTTP(
+		protocol, err := cloudevents.NewHTTP(
 			cloudevents.WithPort(*port),
-			cloudevents.WithPath("/"))
+			cloudevents.WithPath("/"),
+		)
 		if err != nil {
 			return err
 		}
-		client, err := cloudevents.NewClient(transport)
-		if err != nil {
-			return err
+		eventHandler, err := cloudevents.NewHTTPReceiveHandler(ctx, protocol, handler)
```

Comment:
Nice. I did't realize method like `NewHTTPReceiveHandler` exists when implementing panic handler some time ago.

With that I could just call `recoverMiddleware(eventHandler)`, I wouldn't need to do [changes](https://github.com/cloudevents/sdk-go/pull/585) to `sdk-go`, but then again it's probably for best that `sdk` catches that error too.